nav2_costmap_2d/src/footprint.cpp

```diff
@@ -142,13 +142,40 @@ void transformFootprint(
   }
 }
 
+geometry_msgs::msg::Point calculateCentroid(const std::vector<geometry_msgs::msg::Point>& footprint)
+{
+  geometry_msgs::Point center;
+  double sum_x = 0.0, sum_y = 0.0;
+
+  for (const auto& point : footprint)
+  {
+    sum_x += point.x;
+    sum_y += point.y;
+  }
+
+  if (!footprint.empty()) {
+    center.x = sum_x / footprint.size();
+    center.y = sum_y / footprint.size();
+  }
+
+  return center;
+}
+
 void padFootprint(std::vector<geometry_msgs::msg::Point> & footprint, double padding)
 {
-  // pad footprint in place
-  for (unsigned int i = 0; i < footprint.size(); i++) {
-    geometry_msgs::msg::Point & pt = footprint[i];
-    pt.x += sign0(pt.x) * padding;
```

Comment:
Please point out to me the error in the code here - I don't see any immediately. 

Is the issue the case where you have a coordinate exactly at `0.0`? `0.0` for the footprint would be the center of your robot, since that is the `0.0, 0.0` w.r.t. your base_link or base_footprint frame -- which is almost always the center of your robot or aligned with some axle that wouldn't then be a corner of a robot. That would point to me more an issue in your configuration

Reply:
See https://docs.nav2.org/setup_guides/footprint/setup_footprint.html

Reply:
@Nisarg236 should this be closed?